v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/options/OptionsToConfigBuilder.java

```diff
@@ -0,0 +1,97 @@
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.iowrapper.config.TableConfig;
+import com.google.cloud.teleport.v2.source.reader.io.schema.SourceSchemaReference;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public final class OptionsToConfigBuilder {
+
+  public static final class MySql {
+
+    public static JdbcIOWrapperConfig configWithMySqlDefualtsFromOptions(
+        SourceDbToSpannerOptions options) {
+      var builder = builderWithMySqlDefualts();
+      builder =
+          builder
+              .setSourceHost(options.getSourceHost())
+              .setSourcePort(options.getSourcePort())
+              .setSourceSchemaReference(
+                  SourceSchemaReference.builder().setDbName(options.getSourceDB()).build())
+              .setDbAuth(
+                  LocalCredentialsProvider.builder()
+                      .setUserName(options.getUsername())
+                      .setPassword(options.getPassword())
+                      .build())
+              .setJdbcDriverClassName(options.getJdbcDriverClassName())
+              .setJdbcDriverJars(options.getJdbcDriverJars())
+              .setShardID("Unsupported"); /*TODO: Support Sharded Migration */
+      if (options.getSourceConnectionProperties() != "") {
+        builder = builder.setConnectionProperties(options.getSourceConnectionProperties());
+      }
+      if (options.getMaxConnections() != 0) {
+        builder.setMaxConnections((long) options.getMaxConnections());
+      }
+      if (options.getReconnectsEnabled()) {
+        builder.setAutoReconnect(true);
+        if (options.getReconnectAttempts() != 0) {
+          builder.setReconnectAttempts((long) options.getReconnectAttempts());
+        }
+      }
+      var tablesWithPartitionColumns = getTablesWithPartitionColumn(options);
+      ImmutableList<TableConfig> tableConfigs =
+          tablesWithPartitionColumns.entrySet().stream()
+              .map(
+                  entry -> {
+                    var configBuilder =
+                        TableConfig.builder(entry.getKey()).withPartitionColum(entry.getValue());
+                    if (options.getNumPartitions() != 0) {
+                      configBuilder = configBuilder.setMaxPartitions(options.getNumPartitions());
+                    }
+                    if (options.getFetchSize() != 0) {
+                      configBuilder = configBuilder.setMaxFetchSize(options.getFetchSize());
+                    }
+                    return configBuilder.build();
+                  })
+              .collect(ImmutableList.toImmutableList());
+      builder = builder.setTableConfigs(tableConfigs);
+      return builder.build();
+    }
+  }
+
+  private static ImmutableMap<String, String> getTablesWithPartitionColumn(
+      SourceDbToSpannerOptions options) {
+    String[] tables = options.getTables().split(",");
```

Comment:
The tables and partition columns should come from the Primary keys and schema information. Are there scenarios where this config will be required ?

Reply:
Not sure why my reply didn't get recorded with the resolve button.

This PR does not handle auto inference of partition columns (as yet)
I will take it as a separate task before M2.
b/339172915